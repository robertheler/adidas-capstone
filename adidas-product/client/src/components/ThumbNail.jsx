import React, { Component } from 'react';
import styled from 'styled-components';
import fetchImageLocation from '../helper.js';

const RenderDiv = styled.div`
  display: flex;
  flex-direction: column;
  border-radius: 50%;
  width: 35px;
  height: 35px;
  border: ${props => (props.isSelected === true ? '1px solid black' : '1px solid #767677')};
  margin: 0 4px;
  box-sizing: border-box;
  background: url(${props => props.url});
  background-size: 100%;
  -webkit-transition: all 0.2s;
  &:hover {
    border: 1px solid black;
    -webkit-transform: scale(1.35);
  }
`;

class ThumbNail extends Component {
  constructor(props){
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  // Calls the click handler function of the parent component
  // passing the index and url of the clicked ThumbNail
  handleClick(event){
    this.props.handleClick(this.props.index);
  }

  render(){
    return (
      <RenderDiv url={fetchImageLocation(this.props.image)} isSelected={this.props.isSelected} onClick={this.handleClick}>
      </RenderDiv>
    );
  }
}

export default ThumbNail;