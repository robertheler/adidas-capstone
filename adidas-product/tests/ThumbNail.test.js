import React from 'react';
import { shallow, mount, render } from 'enzyme';
import ThumbNail from '../client/src/components/ThumbNail';
import RenderDiv from '../client/src/components/ThumbNail';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0],
    indexOfSelected: 0,
    images: data[0].colors[0].images,
    handleClick: jest.fn()
  };
  const wrapper = shallow(<ThumbNail image={props.images[props.indexOfSelected]} indexOfSelected={props.indexOfSelected} handleClick={props.handleClick}/>);
  return { wrapper, props };
}

describe('Title Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('It should trigger the props color change handler', () => {
    wrapper.instance().handleClick();
    expect(props.handleClick).toHaveBeenCalled();
  });

  it('It should have a black border if selected', () => {
    const divWrapper = <RenderDiv isSelected='true'/>;
    expect(divWrapper).toMatchSnapshot();
  })
});