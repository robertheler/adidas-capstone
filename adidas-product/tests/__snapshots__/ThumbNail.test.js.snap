// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Test Suite It should have a black border if selected 1`] = `
<ThumbNail
  isSelected="true"
/>
`;

exports[`Title Component Test Suite It should render correctly 1`] = `
<styled.div
  onClick={[Function]}
  url="https://adidasproducts.s3-us-west-1.amazonaws.com/images/B42200_color0_pic0.jpg"
/>
`;
